lib/ssz_ex.ex

```diff
@@ -238,28 +236,10 @@ defmodule LambdaEthereumConsensus.SszEx do
     {:ok, root}
   end
 
-  @spec hash_tree_root(struct(), atom()) :: {:ok, Types.root()} | {:error, String.t()}
-  def hash_tree_root(container, module) when is_map(container) do
-    value =
-      module.schema()
-      |> Enum.reduce_while({:ok, <<>>}, fn {key, schema}, {_, acc_root} ->
-        value = container |> Map.get(key)
-
-        case hash_tree_root(value, schema) do
-          {:ok, root} -> {:cont, {:ok, acc_root <> root}}
-          {:error, reason} -> {:halt, {:error, reason}}
-        end
-      end)
-
-    case value do
-      {:ok, chunks} ->
-        leaf_count = chunks |> get_chunks_len() |> next_pow_of_two()
-        root = chunks |> merkleize_chunks_with_virtual_padding(leaf_count)
-        {:ok, root}
-
-      {:error, reason} ->
-        {:error, reason}
-    end
+  def hash_tree_root(vec(_) = list, {:list, _any, _non_neg_integer} = schema) do
+    list
+    |> Aja.Vector.to_list()
```

Comment:
Maybe we can do this as a follow up, but we shouldn't do these kind of things here. We could call this `encode` function (see for example: https://github.com/lambdaclass/lambda_ethereum_consensus/blob/main/lib/types/beacon_chain/beacon_state.ex#L152) before doing the hash tree root in case of a container